lib/util.js

```diff
@@ -398,31 +415,54 @@ function _fetch_rss_page (feed, page, items, all_items, max_items, oldest_date)
         // number of items, cease crawling
         const done_crawling = (limited_items.length < feed_json.items.length);
 
-        if ((is_paginated || is_array) && !done_crawling) {
-            const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
+        const new_all_items = new Set(function* () { yield* all_items; yield* new_items.map(item => item.link); }());
 
-            if (is_paginated)
+        if ((typeof feed_obj !== 'function') && _is_wordpress_feed(feed_json)) {
+            is_paginated = true;
+            const new_feed = _do_create_wordpress(feed_obj);
+            if (is_array) {
+                feed.shift();
+                feed = [new_feed].concat(feed);
+            } else {
+                feed = new_feed;
+            }
+        }
+
+        if (done_crawling) {
+            if (is_paginated && is_array) {
+                feed.shift();
+                return _fetch_rss_page(feed, 1, new_items, new_all_items, max_items, oldest_date);
+            }
```

Comment:
I don't understand this code... If we're done crawling we should not fetch anything else. It may be worth moving the check above, before creating `new_all_items`